### DISCUSSION TITLE
Internationalization of the site

### DISCUSSION THREAD

**Discussion by cedricms** [wrightomated/blood-bowl-roster#157] [Ideas]
Hi,

I was wondering if internationalizing the site ( for example : https://svelte.dev/repl/de39de663ef2445b8fe17b79c500013b?version=3.33.0 ) was something you'd consider?

I'd be more than happy to help out; at the very least for the French translation.

Cheers

**Comment by wrightomated** [comment]
Yes it is something I've wanted to do for a long time.

I have some time this weekend so I might start a bit of a plan. French would be a fantastic second language so assistance with that would be greatly appreciated!

**Reply by cedricms** [reply]
Awesome news James :).

Good luck with the endeavour. Once you've got the language chooser mechanism implemented, feel free to call on me if you want me to handle the i18n of a few components (a "develop" branch would be useful for integration tests in that case).

**Reply by wrightomated** [reply]
Awesome will do!

I've made a start in a feature branch. Just wondering do star players and player positions have translated names too?

**Reply by cedricms** [reply]
Unfortunataly, some of them do.

**Reply by wrightomated** [reply]
Hey @cedricms!

I've set up https://github.com/wrightomated/blood-bowl-roster/tree/feature/158-i18n which you can raise pull requests into. It will kick off a deploy preview and very few tests.

Branch is deployed here: https://feature-158-i18n--bbroster.netlify.app/

I've added a fr.json file https://github.com/wrightomated/blood-bowl-roster/blob/feature/158-i18n/src/locale/fr.json the english file is in the same directory for reference.

ChatGPT did the translation so I'm sure there are many mistakes. I imagine all skill names etc are completely wrong. Just helpful to see what translation keys I was missing.

**Reply by cedricms** [reply]
Hi @wrightomated ,

Thanks for all your hard work!

I translated every element that I had an official GW French translation for (the most recent books are not being translated (Almanacs, Gutter Bowl, ... ) and most French use the English name in that case). ChatGPT did a pretty good job, but didn't match the officials terms on some elements ;).

**Reply by wrightomated** [reply]
Hi Cédric,

Thank you so much for that! So helpful. That is a shame they have stopped translating, I guess it will be easier to add star plays in the future though. I did have a chuckle when it returned Pom-pom girl as cheerleader.

I'll merge it in!
